EIPS/eip-7944.md

```diff
@@ -0,0 +1,194 @@
+---
+eip: 7944
+title: Transaction Ordering,Block-Level Randomness
+description: Dcreasing MEV attacks by forcing a deterministic transaction ordering via block-level randomness.
+author: Aryaethn (@aryaethn)
+discussions-to: https://ethereum-magicians.org/t/potential-eip-mev-decrease-by-deterministic-transaction-ordering-via-block-level-randomness/24084
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-05-24
+---
+
+
+## Abstract
+
+Proposers and builders can currently permute pending transactions arbitrarily, enabling reorder‑driven MEV.  This EIP introduces a consensus rule that sorts all transactions inside a block by XOR‑ing each transaction hash with fresh slot randomness.  The randomness is unknown until the slot starts, so the order is deterministic once known but unpredictable beforehand.  The mechanism **significantly reduces reorder‑based MEV**; latency‑driven back‑running, censorship, and other classes of MEV remain and should be mitigated through complementary techniques (encrypted mempools, reputation, PBS marketplaces, etc.).
+
+## Motivation
+
+Unrestricted ordering is the key enabler of sandwich and classic front‑running attacks.  Deterministic ordering collapses these vectors to latency racing and information asymmetry.  Clear candidate‑set and bundle semantics preserve fee markets while removing the need for trusted sequencers.  Academic works shows deterministic ordering drives sandwich profits toward zero.
+
+### References
+
+* Julia Ofoegbu, “Maximal Extractable Value (MEV): A Tale As Old As Time,” Medium (2024).
+
+* J. Qian et al., “Deterministic Transaction Ordering Without Trusted Sequencers,” arXiv:2411.03327 v1 (2024).
+
+* “Shutter Network Introduces Plan for First Encrypted Mempool on Ethereum,” GlobeNewswire, 13 Feb 2025.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+
+### Slot Randomness `R`
+
+> **Consensus‑layer prerequisite** — Companion EIP “**`EL‑VRF‑exposure`**” is needed to add the RANDAO's per‑slot VRF output to the execution layer.
```

Comment:
is there an EIP for this, if there is then refer here as well as add in required header, if not recommend to add the EIP and then require/refer here